docs/DeveloperGuide.md

```diff
@@ -0,0 +1,115 @@
+# RepoSense - Developer Guide
+Thank you for contributing to RepoSense!
+- [Setting Up](#setting-up)
+- [Architecture](#architecture)
+  - [Parser](#parser)
+  - [Git](#git)
+  - [AuthorshipReporter](#authorshipreporter)
+  - [CommitsReporter](#commitsreporter)
+  - [ReportGenerator](#reportgenerator)
+  - [System](#system)
+  - [Model](#model)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+1. **JDK `1.8.0_60`**  or later.
+2. **Git** on the command line.
+3. **findstr** for Windows, **grep** for macOS or Linux on the command line.
+   * Check that the tool exist on your OS terminal by typing it's name on your terminal and ensure that it does not output messages such as `not found` or `not recognized`.
```

Comment:
`it's` -> `its`

Also, if `Check that the tool....` is meant for point 3 then `tool exist` -> `tool exists`

If you want to ask the user to check for git too then `tool exist` -> `tools exist` 
and use a different indentation. i.e. `>` in markdown